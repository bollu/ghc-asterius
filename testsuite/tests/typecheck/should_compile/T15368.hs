{-# LANGUAGE TypeFamilies #-}

module T15368 where

transitive :: (a, b) -> (b, c) -> (a, c)
transitive = undefined

trigger :: a -> b -> (F a b, F b a)
trigger _ _ = _ `transitive` trigger _ _

type family F (n :: *) (m :: *) :: *
