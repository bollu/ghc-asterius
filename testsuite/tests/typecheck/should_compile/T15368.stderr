
T15368.hs:11:15: warning: [-Wtyped-holes (in -Wdefault)]
    • Found hole: _ :: (F a b, F a0 b0)
      Where: ‘a0’ is an ambiguous type variable
             ‘b0’ is an ambiguous type variable
             ‘a’, ‘b’ are rigid type variables bound by
               the type signature for:
                 trigger :: forall a b. a -> b -> (F a b, F b a)
               at T15368.hs:10:1-35
    • In the first argument of ‘transitive’, namely ‘_’
      In the expression: _ `transitive` trigger _ _
      In an equation for ‘trigger’:
          trigger _ _ = _ `transitive` trigger _ _
    • Relevant bindings include
        trigger :: a -> b -> (F a b, F b a) (bound at T15368.hs:11:1)

T15368.hs:11:15: warning: [-Wdeferred-type-errors (in -Wdefault)]
    • Couldn't match type ‘F b a’ with ‘F b0 a0’
      Expected type: (F a b, F b a)
        Actual type: (F a b, F b0 a0)
      NB: ‘F’ is a non-injective type family
      The type variables ‘b0’, ‘a0’ are ambiguous
    • In the expression: _ `transitive` trigger _ _
      In an equation for ‘trigger’:
          trigger _ _ = _ `transitive` trigger _ _
    • Relevant bindings include
        trigger :: a -> b -> (F a b, F b a) (bound at T15368.hs:11:1)

T15368.hs:11:38: warning: [-Wtyped-holes (in -Wdefault)]
    • Found hole: _ :: a0
      Where: ‘a0’ is an ambiguous type variable
    • In the first argument of ‘trigger’, namely ‘_’
      In the second argument of ‘transitive’, namely ‘trigger _ _’
      In the expression: _ `transitive` trigger _ _
    • Relevant bindings include
        trigger :: a -> b -> (F a b, F b a) (bound at T15368.hs:11:1)

T15368.hs:11:40: warning: [-Wtyped-holes (in -Wdefault)]
    • Found hole: _ :: b0
      Where: ‘b0’ is an ambiguous type variable
    • In the second argument of ‘trigger’, namely ‘_’
      In the second argument of ‘transitive’, namely ‘trigger _ _’
      In the expression: _ `transitive` trigger _ _
    • Relevant bindings include
        trigger :: a -> b -> (F a b, F b a) (bound at T15368.hs:11:1)
